src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,759 @@
+    (recordings: ArchivedRecording[]) => {
+      const freshestRecording = recordings.reduce((prev, current) =>
+        prev?.archivedTime > current?.archivedTime ? prev : current
+      );
+      addSubscription(
+        context.target
+          .target()
+          .pipe(first())
+          .subscribe((target) => {
+            context.reports
+              .reportJson(freshestRecording, target.connectUrl)
+              .pipe(first())
+              .subscribe({
+                next: (report) => {
+                  setReport(freshestRecording.name);
+                  setUsingArchivedReport(true);
+                  setReportTime(freshestRecording.archivedTime);
+                  categorizeEvaluation(report);
+                  setIsLoading(false);
+                },
+                error: (err) => {
+                  handleStateErrors(err.message);
+                },
+              });
+          })
+      );
+    },
+    [
+      addSubscription,
+      context.target,
+      context.reports,
+      categorizeEvaluation,
+      handleStateErrors,
+      setUsingArchivedReport,
+      setReportTime,
+      setIsLoading,
+      setReport,
+    ]
+  );
+
+  // try generating report on cached or archived recordings
+  const handleEmptyRecordings = React.useCallback(
+    (connectUrl: string) => {
+      const cachedReportAnalysis = context.reports.getCachedAnalysisReport(connectUrl);
+      if (cachedReportAnalysis.report.length > 0) {
+        setReport('automated-analysis');
+        setUsingCachedReport(true);
+        setReportTime(cachedReportAnalysis.timestamp);
+        categorizeEvaluation(cachedReportAnalysis.report);
+        setIsLoading(false);
+      } else {
+        addSubscription(
+          queryArchivedRecordings(connectUrl)
+            .pipe(
+              first(),
+              map((v) => v.data.archivedRecordings.data as ArchivedRecording[])
+            )
+            .subscribe({
+              next: (recordings) => {
+                if (recordings.length > 0) {
+                  handleArchivedRecordings(recordings);
+                } else {
+                  handleStateErrors(NO_RECORDINGS_MESSAGE);
+                }
+              },
+              error: (err) => {
+                handleStateErrors(err.message);
+              },
+            })
+        );
+      }
+    },
+    [
+      addSubscription,
+      context.reports,
+      categorizeEvaluation,
+      queryArchivedRecordings,
+      handleArchivedRecordings,
+      handleStateErrors,
+      setUsingCachedReport,
+      setReportTime,
+      setReport,
+      setIsLoading,
+    ]
+  );
+
+  const generateReport = React.useCallback(() => {
+    addSubscription(
+      context.target
+        .target()
+        .pipe(
+          filter((target) => target !== NO_TARGET),
+          first()
+        )
+        .subscribe((target) => {
+          handleLoading();
+          setTargetConnectURL(target.connectUrl);
```

Comment:
Remove line `setTargetConnectURL` and `dispatch` here. Already handled in `context.target.target` when target changes

Reply:
Should be line 327 and 328